ci-operator/step-registry/sandboxed-containers-operator/create-prowjob/sandboxed-containers-operator-create-prowjob-commands.sh

```diff
@@ -0,0 +1,418 @@
+if ! [[ "${TEST_TIMEOUT}" =~ ^[0-9]+$ ]]; then
+    echo "ERROR: TEST_TIMEOUT should be numeric, got: ${TEST_TIMEOUT}"
+    exit 1
+fi
+
+
+
+# Catalog Source Configuration
+echo "Configuring catalog sources..."
+
+# Set catalog source variables based on TEST_RELEASE_TYPE
+if [[ "${TEST_RELEASE_TYPE}" == "Pre-GA" ]]; then
+    # OSC Catalog Configuration - get latest or use provided
+    if [[ -z "${OSC_CATALOG_TAG:-}" ]]; then
+        OSC_CATALOG_TAG=$(get_latest_osc_catalog_tag)
+    else
+        echo "Using provided OSC_CATALOG_TAG: ${OSC_CATALOG_TAG}"
+    fi
+
+    CATALOG_SOURCE_IMAGE="${CATALOG_SOURCE_IMAGE:-quay.io/redhat-user-workloads/ose-osc-tenant/osc-test-fbc:${OSC_CATALOG_TAG}}"
+    CATALOG_SOURCE_NAME="${CATALOG_SOURCE_NAME:-brew-catalog}"
+
+    # Trustee Catalog Configuration
+    # Convert OCP version for Trustee catalog naming
+    OCP_VER=$(echo "${OCP_VERSION}" | tr '.' '-')
+    subfolder=""
+    if [[ "${OCP_VER}" == "4-16" ]]; then
+        subfolder="trustee-fbc/"
+    fi
+    # Get latest Trustee catalog tag with page limit safety
+    TRUSTEE_REPO_NAME="${subfolder}trustee-fbc-${OCP_VER}"
+    TRUSTEE_CATALOG_REPO="quay.io/redhat-user-workloads/ose-osc-tenant/${TRUSTEE_REPO_NAME}"
+
+    APIURL="https://quay.io/api/v1/repository/redhat-user-workloads/ose-osc-tenant/${TRUSTEE_REPO_NAME}"
+    TRUSTEE_CATALOG_TAG=$(get_latest_trustee_catalog_tag)
+
+    TRUSTEE_CATALOG_SOURCE_IMAGE="${TRUSTEE_CATALOG_SOURCE_IMAGE:-${TRUSTEE_CATALOG_REPO}:${TRUSTEE_CATALOG_TAG}}"
+    TRUSTEE_CATALOG_SOURCE_NAME="${TRUSTEE_CATALOG_SOURCE_NAME:-trustee-catalog}"
+else # GA
+    CATALOG_SOURCE_NAME="${CATALOG_SOURCE_NAME:-redhat-operators}"
+    TRUSTEE_CATALOG_SOURCE_NAME="${TRUSTEE_CATALOG_SOURCE_NAME:-redhat-operators}"
+fi
+
+# Generate output file path
+OCP_PROWJOB_VERSION=$(echo "${OCP_VERSION}" | tr -d '.' )
+OUTPUT_FILE="openshift-sandboxed-containers-operator-devel__downstream-${PROW_RUN_TYPE}${OCP_PROWJOB_VERSION}.yaml"
+echo "Generating prowjob configuration..."
+
+# Backup existing file if it exists
+if [[ -f "${OUTPUT_FILE}" ]]; then
+    echo "Backing up existing file: ${OUTPUT_FILE}.backup"
+    cp "${OUTPUT_FILE}" "${OUTPUT_FILE}.backup"
+fi
+
+# Generate the prowjob configuration file
+
+cat > "${OUTPUT_FILE}" <<EOF
+documentation: |-
+  DO NOT EDIT DIRECTLY.
+  This is generated by the sandboxed-containers-operator-create-prowjob-commands.sh script.
+base_images:
+  tests-private:
+    name: tests-private
+    namespace: ci
+    tag: "4.20"
+  upi-installer:
+    name: "${OCP_VERSION}"
+    namespace: ocp
+    tag: upi-installer
+releases:
+  latest:
+    release:
+      architecture: amd64
+      channel: stable
+      version: "${OCP_VERSION}"
+resources:
+  '*':
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: azure-ipi-kata
+  cron: 0 0 31 2 1
+  steps:
+    cluster_profile: azure-qe
+    env:
+      CUSTOM_AZURE_REGION: ${CUSTOM_AZURE_REGION}
+      BASE_DOMAIN: qe.azure.devcluster.openshift.com
```

Comment:
IMO it'd be better to make these configurable by env variables as well... Or, actually, how about create a function to generate the shared parts rather then repeating them multiple times?